code/script.js

```diff
@@ -303,41 +323,85 @@ const filterCharacters = (keep) => {
         `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
       )
     }
-  } else if (group === 'other') {
+  } else if (category === 'eye color') {
+     if (keep) {
+      alert(
+        `Yes, the person have ${value} eyes! Keep all that have ${value} eyes!`
+      )
+    } else {
+      alert(
+        `No, the person do not have ${value} eyes! Remove all that have ${value} eyes!`
+      )
+    }
+  }
+  
+  else if (category === 'other') {
+    if (keep) {
+      alert(
+        `Yes, the person smokes! Keep all persons that smoke`
+      )
+    } else {
+      alert(
+        `No, the person doesn't smoke! Remove all persons that smoke`
+      )
+    }
     // Similar to the one above
-  } else {
+  } 
+  else if (category === 'hair color') {
     if (keep) {
+      alert(`Yes, the person has ${value} hair! Keep all persons with ${value} hair`)
       // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
     } else {
+      alert(`No, the person doesn't have ${value} hair. Remove all persons with ${value} hair`)
       // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+    } 
   }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+ generateBoard()
 
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
+  const isConfirmed = confirm(`Are you sure you want to make a guess on ${suspect}?`)
+  if (isConfirmed) {
+    checkMyGuess(suspect)
+  }
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = `You guessed on ${suspect} and it was correct!`
+  } else {
+    winOrLoseText.innerHTML = `Wrong answer. It was actually ${secret.name}!`
+  }
+  console.log("suspect", suspect)
+  console.log("secret name", secret.name)
```

Comment:
When you're finished with the project it's nice to clean out the console.log. They are great for development but not needed in production :)